src/NATS.Client.JetStream/NatsJSOrderedConsumer.cs

```diff
@@ -199,6 +199,32 @@ await foreach (var msg in consumer.FetchAsync(opts, serializer, cancellationToke
             _fetchConsumerName = string.Empty;
     }
 
+    /// <inheritdoc />
+    public async IAsyncEnumerable<NatsJSMsg<T>> FetchNoWaitAsync<T>(
+        NatsJSFetchOpts opts,
+        INatsDeserialize<T>? serializer = default,
+        [EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        cancellationToken = CancellationTokenSource.CreateLinkedTokenSource(_cancellationToken, cancellationToken).Token;
+
+        var consumer = await RecreateConsumer(_fetchConsumerName, _fetchSeq, cancellationToken);
+        _fetchConsumerName = consumer.Info.Name;
+
+        await foreach (var msg in consumer.FetchNoWaitAsync(opts, serializer, cancellationToken))
+        {
+            if (msg.Metadata is not { } metadata)
+                continue;
+
+            _fetchSeq = metadata.Sequence.Stream;
+            yield return msg;
+        }
+
+        var deleted = await TryDeleteConsumer(_fetchConsumerName, cancellationToken);
```

Comment:
should `TryDeleteConsumer` be run inside of a `finally` - for example if the `CancellationToken` throws an `OperationCanceledException`?